flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-kafka/src/main/java/org/apache/flink/cdc/connectors/kafka/utils/JsonRowDataSerializationSchemaUtils.java

```diff
@@ -0,0 +1,94 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.cdc.connectors.kafka.utils;
+
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.formats.common.TimestampFormat;
+import org.apache.flink.formats.json.JsonFormatOptions;
+import org.apache.flink.formats.json.JsonRowDataSerializationSchema;
+import org.apache.flink.table.types.logical.RowType;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+
+/** Utils for creating JsonRowDataSerializationSchema. */
+public class JsonRowDataSerializationSchemaUtils {
+
+    /**
+     * In flink>=1.20, the constructor of JsonRowDataSerializationSchema has 6 parameters, and in
+     * flink<1.20, the constructor of JsonRowDataSerializationSchema has 5 parameters.
+     */
+    public static JsonRowDataSerializationSchema createSerializationSchema(
+            RowType rowType,
+            TimestampFormat timestampFormat,
+            JsonFormatOptions.MapNullKeyMode mapNullKeyMode,
+            String mapNullKeyLiteral,
+            boolean encodeDecimalAsPlainNumber,
+            boolean ignoreNullFields) {
+        try {
+            Class<?>[] fullParams =
+                    new Class[] {
+                        RowType.class,
+                        TimestampFormat.class,
+                        JsonFormatOptions.MapNullKeyMode.class,
+                        String.class,
+                        boolean.class,
+                        boolean.class
+                    };
+
+            Object[] fullParamValues =
+                    new Object[] {
+                        rowType,
+                        timestampFormat,
+                        mapNullKeyMode,
+                        mapNullKeyLiteral,
+                        encodeDecimalAsPlainNumber,
+                        ignoreNullFields
+                    };
+
+            for (int i = fullParams.length; i >= 5; i--) {
+                try {
+                    Constructor<?> constructor =
+                            JsonRowDataSerializationSchema.class.getConstructor(
+                                    Arrays.copyOfRange(fullParams, 0, i));
+
+                    return (JsonRowDataSerializationSchema)
+                            constructor.newInstance(Arrays.copyOfRange(fullParamValues, 0, i));
+                } catch (NoSuchMethodException ignored) {
+                }
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to create JsonRowDataSerializationSchema", e);
+        }
+        throw new RuntimeException(
+                "Failed to find appropriate constructor for JsonRowDataSerializationSchema");
```

Comment:
Add some comments like "Please check that your Flink version is 1.19 or 1.20" to guide user?